src/adapters/ui/iconview.js

```diff
@@ -322,11 +316,13 @@ export class DesktopIconView extends EventEmitter {
           .then(entries => actions.setEntries(entries));
       }
 
-    }, view(fileIcon, themeIcon, droppable), this.$root);
+    }, view(this.core, fileIcon, themeIcon, droppable), this.$root);
 
     this.applySettings();
     this.iconview.reload();
     this._createWatcher();
+
+    this.core.on('osjs/settings:save', this.iconview.reload);
```

Comment:
Make arguments explicit here as events might send data and make for horrible side-effects :D

```
    this.core.on('osjs/settings:save', () => this.iconview.reload());
```